bcpandas/main.py

```diff
@@ -441,7 +442,20 @@ def to_sql(
     # save to temp path
     csv_file_path = get_temp_file(work_directory)
     # replace bools with 1 or 0, this is what pandas native does when writing to SQL Server
-    df.replace({True: 1, False: 0}).to_csv(
+    # attention: the `(lambda col: lambda...)(copy(col))` part looks odd but is
+    # needed to ensure loop iterations create lambdas working on different columns.
+    df_out = df.assign(
+        **{
+            col: (lambda col: lambda df: df[col].map({True: 1, False: 0}).astype(pd.Int8Dtype()))(
+                copy(col)
+            )
```

Comment:
This reads a little strange, can't we do this eagerly?

Reply:
It is strange. But otherwise the col variable is reused within multiple lambda expressions and thus it would always reference the last column. We could do it non-functionally with a for loop.

Reply:
I think the functional df.assign method is designed to work with lambdas. As I said, we could do this non-functionally.

Reply:
```suggestion
            col: lambda df, col=col: df[col].map({True: 1, False: 0}).astype(pd.Int8Dtype())
```